src/containers/trends/TrendsContainer.js

```diff
@@ -154,48 +174,55 @@ export class TrendsContainer extends React.Component {
   }
 
   getCurrentDay() {
+    const { timePrefs } = this.props;
     const { dateDomain: { end } } = this.state;
-    // TODO: replace with more robust code for finding noon of the local timezone day
-    return new Date(Date.parse(end) - (864e5 / 2)).toISOString();
+    return datetime.localNoonBeforeTimestamp(
+      end,
+      datetime.getTimezoneFromTimePrefs(timePrefs)
+    ).toISOString();
   }
 
   setExtent(newDomain) {
     const { cbgByDate, smbgByDate } = this.props;
+    const { mostRecent } = this.state;
     this.refilterByDate(cbgByDate, newDomain);
     this.refilterByDate(smbgByDate, newDomain);
     this.setState({
       currentCbgData: cbgByDate.top(Infinity).reverse(),
       currentSmbgData: smbgByDate.top(Infinity).reverse(),
       dateDomain: { start: newDomain[0], end: newDomain[1] },
```

Comment:
there is no possibility that there aren't two dates here? could that be an issue if there aren't?